core/driver/sources/src/main/scala/org/virtuslab/ideprobe/config/IntelliJProvider.scala

```diff
@@ -0,0 +1,93 @@
+package org.virtuslab.ideprobe.config
+
+import com.google.gson.Gson
+
+import java.nio.file.{CopyOption, FileVisitResult, Files, Path, SimpleFileVisitor}
+import org.virtuslab.ideprobe.ide.intellij.{InstalledIntelliJ, IntelliJFactory, LocalIntelliJ}
+import org.virtuslab.ideprobe.IdeProbePaths
+import org.virtuslab.ideprobe.dependencies.{IntelliJVersion, Plugin}
+
+import java.nio.file.attribute.BasicFileAttributes
+import scala.io.Source
+
+trait IntelliJProvider {
+  def factory: IntelliJFactory
+  def setup(): InstalledIntelliJ
+  def version: IntelliJVersion
+  def withFactory(factory: IntelliJFactory): IntelliJProvider
+}
+
+case class ExistingIntelliJ(
+  path: Path,
+  plugins: Seq[Plugin],
+  factory: IntelliJFactory
+) extends IntelliJProvider {
+  override def setup(): InstalledIntelliJ = {
+    val pluginsDir = path.resolve("plugins")
+    val backupDir = Files.createTempDirectory(path, "plugins")
+    copyFolder(pluginsDir, backupDir)
+    factory.installPlugins(plugins, path)
+
+    new LocalIntelliJ(path, factory.paths, factory.config, backupDir)
+  }
+
+  private def copyFolder(source: Path, target: Path): Unit = {
```

Comment:
put it in ProbeExtensions?

Reply:
Such method probably already exists in PathExtension

Reply:
Right. Removed.